OvmfPkg/OvmfXen: Use RuntimeTimerLibCpu for DXE_DRIVER

When booting Windows, it's still possible that something in Windows bootloader
tries to use TimerLib with TPL already at TPL_HIGH_LEVEL, which ultimately
tries to load the dynamic PcdFSBClock. And this fails because PcdGet() tries
to raise TPL to a lower value.

This allow to load the PCD early, at library construction time, and thus avoid
the need to get the PCD when the TPL is too high.

Signed-off-by: Anthony PERARD <anthony.perard@citrix.com>
---

This TimerLib could be at DXE_RUNTIME_DRIVER, UEFI_APPLICATION, UEFI_DRIVER,
but is that useful?

diff --git a/OvmfPkg/OvmfXen.dsc b/OvmfPkg/OvmfXen.dsc
index 01dac02f6d..5f2e1dc9aa 100644
--- a/OvmfPkg/OvmfXen.dsc
+++ b/OvmfPkg/OvmfXen.dsc
@@ -358,6 +358,7 @@ [LibraryClasses.common.DXE_DRIVER]
   PciLib|OvmfPkg/Library/DxePciLibI440FxQ35/DxePciLibI440FxQ35.inf
   MpInitLib|UefiCpuPkg/Library/MpInitLib/DxeMpInitLib.inf
   QemuFwCfgS3Lib|OvmfPkg/Library/QemuFwCfgS3Lib/DxeQemuFwCfgS3LibFwCfg.inf
+  TimerLib|MdePkg/Library/SecPeiDxeTimerLibCpu/RuntimeTimerLibCpu.inf
 
 [LibraryClasses.common.UEFI_APPLICATION]
   PcdLib|MdePkg/Library/DxePcdLib/DxePcdLib.inf
